File: Checks/PWR004/example-omp.c
// PWR004: Declare OpenMP scoping for all variables

void example(int *result, unsigned size) {
  int factor = 42;

  // No data scoping is specified
  #pragma omp parallel for
  for (int i = 0; i < size; i++) {
    result[i] = factor * i;
  }
}

File: Checks/PWR004/example-omp.f90
! PWR004: Declare OpenMP scoping for all variables

subroutine example(result)
  implicit none
  integer, intent(out) :: result(:)
  integer :: i, factor
  
  factor = 42

  ! No data scoping is specified
  !$omp parallel do
  do i = 1, size(result, 1)
    result(i) = factor * i
  end do
end subroutine example
